src/test/java/uk/gov/pay/webhooks/webhook/resource/WebhookSigningKeyIT.java

```diff
@@ -0,0 +1,74 @@
+package uk.gov.pay.webhooks.webhook.resource;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import uk.gov.pay.extension.AppWithPostgresExtension;
+import uk.gov.pay.webhooks.util.DatabaseTestHelper;
+
+import javax.ws.rs.core.Response;
+import java.util.Map;
+
+import static io.restassured.RestAssured.given;
+import static io.restassured.http.ContentType.JSON;
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.CoreMatchers.startsWith;
+import static org.hamcrest.Matchers.hasLength;
+import static org.hamcrest.Matchers.not;
+
+public class WebhookSigningKeyIT {
+    @RegisterExtension
+    public static AppWithPostgresExtension app = new AppWithPostgresExtension();
+    private Integer port = app.getAppRule().getLocalPort();
+    private DatabaseTestHelper dbHelper;
+
+    @BeforeEach
+    public void setUp() {
+        dbHelper = DatabaseTestHelper.aDatabaseTestHelper(app.getJdbi());
+        dbHelper.truncateAllData();
+    }
+    
+    @Test
+    public void shouldGetThenResetSigningKey() {
+        var json = """
+                {
+                  "service_id": "test_service_id",
+                  "live": true,
+                  "callback_url": "https://example.com",
+                  "description": "description",
+                  "subscriptions": ["card_payment_captured"]
+                }
+                """;
+        var response = given().port(port)
+                .contentType(JSON)
+                .body(json)
+                .post("/v1/webhook")
+                .then()
+                .statusCode(Response.Status.OK.getStatusCode())
+                .extract()
+                .as(Map.class);
+
+        var externalId = response.get("external_id");
+        var serviceId = response.get("service_id");
+        
+        var signingKeyResponse = given().port(port)
+                .contentType(JSON)
+                .get("/v1/webhook/%s/signing-key?service_id=%s".formatted(externalId, serviceId))
+                .then()
+                .statusCode(Response.Status.OK.getStatusCode())
+                .body("signing_key", startsWith("webhook_live"))
+                .body("signing_key", hasLength(39))
+                .extract()
+                .as(Map.class);
+
+        var originalSigningKey = signingKeyResponse.get("signing_key");
+
+        given().port(port)
+                .contentType(JSON)
+                .post("/v1/webhook/%s/signing-key?service_id=%s".formatted(externalId, serviceId))
+                .then()
+                .statusCode(Response.Status.OK.getStatusCode())
+                .body("signing_key", not(originalSigningKey))
+                .body("signing_key", hasLength(39));
+    }
```

Comment:
Possibly we should finish this with another GET to check that that the we get the new signing key again so we know it’s definitely been persisted etc.?

Reply:
added